homeassistant/components/prometheus/__init__.py

```diff
@@ -355,13 +362,21 @@ def state_as_number(state: State) -> float:
             value = 0
         return value
 
-    @staticmethod
-    def _labels(state: State) -> dict[str, Any]:
-        return {
+    def _labels(self, state: State) -> dict[str, Any]:
+        final_labels = {
             "entity": state.entity_id,
             "domain": state.domain,
             "friendly_name": state.attributes.get(ATTR_FRIENDLY_NAME),
```

Comment:
This is not something that is directly related to what you are doing here, but since many labels will be changing here, I'd suggest to have `friendly_name` default to an empty string instead of "None", e.g.

```
"friendly_name": state.attributes.get(ATTR_FRIENDLY_NAME) or "",
```